### DISCUSSION TITLE
Very Slow Download Speeds for Sources on Github

### DISCUSSION THREAD

**Discussion by drankinatty** [community/community#143212] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I have had declining downloads speeds from all github project over the past few months. My ISP is Optimum with the account located in Texas. I have a 300Mbs up/25Mbs down connection with fixe IPv4 address. The download speeds I get from github a basically dial-up speeds of less than 100Kbps.

For example, the last download of [mongodb 4.4.29 source](https://github.com/mongodb/mongo/archive/refs/tags/r4.4.29.tar.gz) took nearly 10 minutes to download 50MB, e.g.

```none
  -> Downloading r4.4.29.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 49.2M    0 49.2M    0     0  90525      0 --:--:--  0:09:30 --:--:--  149k
```

The individual hop-times seem okay in `traceroute` between my computer and `github.com`, but there are 30 hops listed, e.g.

```none
# traceroute github.com
traceroute to github.com (140.82.114.4), 30 hops max, 60 byte packets
 1  _gateway (192.168.6.11)  0.751 ms  1.021 ms  0.700 ms
 2  * * *
 3  173.219.152.4 (173.219.152.4)  13.881 ms  13.866 ms  14.342 ms
 4  173.219.197.86 (173.219.197.86)  17.576 ms  17.693 ms  17.549 ms
 5  dls-b23-link.ip.twelve99.net (213.248.88.228)  16.846 ms  16.832 ms  19.765 ms
 6  dls-bb1-link.ip.twelve99.net (62.115.136.118)  20.769 ms * *
 7  * * *
 8  * atl-bb1-link.ip.twelve99.net (62.115.137.54)  34.364 ms  35.094 ms
 9  rest-bb1-link.ip.twelve99.net (62.115.138.70)  53.207 ms  54.298 ms  53.931 ms
10  rest-b2-link.ip.twelve99.net (62.115.123.41)  53.472 ms  52.949 ms  50.715 ms
11  github-ic-368832.ip.twelve99-cust.net (213.248.67.47)  64.777 ms  50.298 ms  53.864 ms
12  * * *
13  * * *
14  * * *
15  * * *
16  * * *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
26  * * *
27  * * *
28  * * *
29  * * *
30  * * *
```

Is there a general problem with the download bandwidth available for github users or does this point to a technical issue somewhere in the hops between Texas and wherever the github servers are located?

**Comment by if-i-shajan** [comment]
Is this a problem? I had ignore this for long time. Please anyone help to solve the issues.

**Comment by digitalcohesion** [comment]
It looks like traffic to github is being routed through twelve99 which appears to be a company based in Sweden. Not sure where the traffic is actually being routed to, but since this was posted 2 days ago and it's still a problem my confidence in whoever is running network operations at github is slightly diminished at this point. I just noticed it this morning trying to download a file that would ordinarily take a minute or two and now it's showing 3 hours.

I tried to find a place to submit a ticket to anything remotely resembling network operations and couldn't find anything so I ended up here. Oh well. I guess we just wait until someone important or connected enough notices and gets it resolved.

**Comment by bathulababu** [comment]
Response#1

**Comment by iirekm** [comment]
I'm in Poland, ISP is UPC/Play fiber, and similar issue. Download speeds are about 0.5 MiB/s, while for other websites are tens of MiB/s. Downloading bigger things is a problem because it seems there's some timeout on github side - after few minutes download link seems to expire! What is going on?

**Reply by iirekm** [reply]
What are global addresses of github servers? Maybe manually setting a close one in /etc/hosts fixes the issue?

**Comment by jfullstackdev** [comment]
I never had any issues, even using the GitHub Desktop cloning repos and maybe you can try GitHub Desktop too, to see if there is any difference,

also, have you tried downloading it as `.zip` from GitHub, to again, just test the behavior ? it's in the Code tab

<img width="442" height="351" alt="image" src="https://github.com/user-attachments/assets/67826afd-08d2-4013-81e2-9450d46bbf88" />

and likely the issues are : 

1. **Network routing issues** – Traffic to GitHub may be taking inefficient paths (e.g., routed via *twelve99* in Sweden), causing high latency and slow speeds.
2. **ISP bandwidth throttling or congestion** – Your provider might limit speeds during large file transfers or at peak times.
3. **Temporary regional GitHub CDN slowdown** – GitHub’s content delivery network could be experiencing performance issues in your area.
4. **Local network limitations** – Router, firewall, or Wi-Fi bottlenecks affecting sustained transfer speeds.
5. **Unusually large repository size** – Large file history or binaries in the repo can make cloning inherently slower.

**Reply by iirekm** [reply]
I'm not talking about cloning git repositories, I'm talking about downloading of artifacts from public repos, e.g. compiled elf, exe or msi files. Some of them are few hundred megabytes in size, and downloading them in my location is nightmare (slow, ends with errors, download links expire - can't be resumed and have start from scratch).
It's not my local network, it's not my network provider - when I tried different provider or different laptop -exactly the same problem. It's something very, veeery wrong with GitHub in Warsaw. We have lots of CDNs here, we have Azure datacenter here, looks like something gets routed terribly.

**Reply by iirekm** [reply]
When I run traceroute I get:
```
➜  ~ traceroute release-assets.githubusercontent.com
traceroute to release-assets.githubusercontent.com (185.199.110.133), 30 hops max, 60 byte packets
...
 6  pl-waw09e-ri1_ae2.0.as9141.pl (185.182.244.27)  25.837 ms  29.089 ms *
 7  war-b3-link.ip.twelve99.net (62.115.170.122)  29.088 ms  29.547 ms  36.486 ms
 8  ffm-bb2-link.ip.twelve99.net (62.115.120.20)  54.914 ms  35.258 ms *
 9  ffm-b17-link.ip.twelve99.net (62.115.132.115)  56.009 ms ffm-b11-link.ip.twelve99.net (62.115.124.119)  34.784 ms  39.565 ms
 ```
 
 Yes - this goes via this unfortunate twelve99. How to use a different route?

**Reply by iirekm** [reply]
Also: dig release-assets.githubusercontent.com - why it points to IP addresses in Los Angeles? Doesn't GitHub have CDNs in Europe???

**Comment by minifisch** [comment]
I have similar issues on my end. I have only one Internet Exchange between my ISP and Fastly, where GitHub servers are located.
Many projects are moving to ghcr.io and therefore I am forced to use GitHubs Container Registry. The download speed for the images are roughly between 20-40 MBit/s. 

Please see this traceroute - I cant imagine that my peering is an issue here:
```cachy (10.0.0.30) -> cdn-185-199-111-154.github.com (185.199.111.154)                                                              2025-09-27T14:02:02+0200
Keys:  Help   Display mode   Restart statistics   Order of fields   quit
                                                                                                                   Packets               Pings
 Host                                                                                                            Loss%   Snt   Last   Avg  Best  Wrst StDev
 1. 10.0.0.1                                                                                                      0.0%   422    0.1   0.1   0.1   2.1   0.1
 2. 100.126.0.244                                                                                                 0.0%   422    3.0   5.2   2.5  65.0   9.2
 3. 100.126.0.46                                                                                                  0.0%   422    3.1   5.0   2.5  62.2   8.7
 4. 193.203.0.66                                                                                                  0.0%   421    3.6   3.2   2.7   4.3   0.3
 5. 185.199.111.154                                                                                               0.0%   421    3.4   3.4   2.7   9.0   0.5
```

193.203.0.66 is Fastly Inc. at VIX and  185.199.111.154 belongs to GitHub via Fastly.

I got a 500 MBit/s downlink from my ISP and all the other content is coming with the exact line speed.
This is really frustrating.

**Reply by tinsever** [reply]
Still is!

**Comment by tinsever** [comment]
For me using a VPN near SF, USA worked. Seems like GitHub CDN is .... bad 🤷🏻

**Comment by daalfox** [comment]
And once again I'm sitting here cloning a small repo for 2 hours. Thank you Microsoft!

**Comment by phraemer** [comment]
My office and my home which are physically close in the same city use the same ISP but at home I have the slow download speeds from GitHub too.
Tried changing DNS but didn't help.

**Reply by tinsever** [reply]
It's dependant on IP routing not on DNS, sadly. And that can vary also while using the same ISP.

**Reply by phraemer** [reply]
Ahh that's unfortunate. Thanks for the info. I wonder if there is someone nice at GitHub/Microsoft that can look at things like this.

**Comment by T4ke-Github** [comment]
I have the same problem in Germany / Berlin. Abysmall slow download speed with my 500 Mbps line. Changing to Norway via VPN fix that problem. Download speeds are jumping back to ~400 - 500 Mbps. Without VPN I'm stuck with 100 kbps.

**Comment by bnussman** [comment]
We good @github?

I don't want to have to wait 50 minutes to setup a Github Actions runner... As @minifisch mentioned GitHubs Container Registry is also affected by this.

I'm on ATT Fiber 500/500 

<img width="544" height="99" alt="Screenshot 2026-01-22 at 9 44 05 PM" src="https://github.com/user-attachments/assets/adedf5b3-5db2-4890-96c3-951639584e05" />

Here's a traceroute if that helps...

```sh
traceroute to ghcr.io (140.82.112.34), 64 hops max, 40 byte packets
 1  192.168.0.1 (192.168.0.1)  1.309 ms  0.550 ms  0.338 ms
 2  192.168.1.254 (192.168.1.254)  0.949 ms  0.943 ms  0.924 ms
 3  104-186-148-1.lightspeed.rlghnc.sbcglobal.net (104.186.148.1)  1.552 ms  1.534 ms  5.043 ms
 4  99.173.77.18 (99.173.77.18)  3.946 ms  3.220 ms  3.886 ms
 5  * * *
 6  * * *
 7  * * *
 8  192.205.32.178 (192.205.32.178)  12.323 ms  11.452 ms  13.038 ms
 9  ce-0-2-0.a05.asbnva02.us.ce.gin.ntt.net (157.238.226.211)  10.884 ms  12.421 ms  11.856 ms
10  * * *
11  * * *
12  * * *
13  * * *
14  * * *
15  * * *
16  * * *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
```

**Reply by bnussman** [reply]
Seems to be fixed for me now.

**Comment by tinsever** [comment]
Yep. This looks like an routing issue between you and GitHub's infra. You can see that there are many ICMP package drops – maybe try a proxy?

**Comment by nipsy** [comment]
I think I'm running into this exact same problem as described [here](https://github.com/NixOS/nix/issues/10683#issuecomment-3865154540).  This has been increasingly more problematic over the last several months to the point where now, I'm unable to fully download artifacts from github on a semi-regular basis.  Good times.

*edit*

I'll include what I found in my linked issue above.  The current rtw89_8852ce driver is apparently garbage and causes weird connection issues.  I removed it entirely from the equation and my desktop can transfer from GitHub at a normal speed again (in line with my 1Gbps fiber).

**Comment by iirekm** [comment]
Can someone from Micro$oft eventually look at it? In recent months you broke windows (every update is a nightmare), and you broke github (every download from here in Europe in nightmare because goes to Los Angeles)!
Tried `traceroute release-assets.githubusercontent.com` also today - the same issues as months ago!

**Comment by DavidPriceNBS** [comment]
I'm experiencing this issue now across all self-hosted github runners trying to pull from the github package registry, taking 10 minutes to download before the runner even starts to process the job.

**Reply by ansxor** [reply]
This just started for me today. Many of my GitHub actions that would pull from the GHCR are now failing from the job timing out after 10 minutes when downloading. Ridiculous. Not to mention this is affecting our other servers using ArgoCD.

**Reply by DavidPriceNBS** [reply]
I don't mind there being issues, inevitable part of the complexity of what they provide, but I find it odd that there was no report of degradation whatsoever on their status page yesterday or today. I find quite often people say it to try to get a reaction or reply or "stick it to" the company - but genuinely I think we'll need to move our runner image hosting elsewhere. Created real blockers for us yesterday having every image pull - from both my local machine and from my runner containers - taking 10+ minutes to pull the image down.

**Comment by damchey** [comment]
I am experiencing this problem for that last six months. Two connections, same ISP, almost same traceroute, but one connection is getting  only 1Mbps and the other is getting line speed. Only difference is these two connections have different IP blocks/subnets.

Have a feeling github/azure/MS is throttling/rate-limiting connections based on IPs.

Below is the result of github-debug.com on the problem connection:

```
2507998 bytes downloaded from github.com at 1.13 Mbps
2507998 bytes downloaded from cloud.githubusercontent.com at 2.73 Mbps
2507998 bytes downloaded from avatars.githubusercontent.com at 42.15 Mbps
2507998 bytes downloaded from github.githubassets.com at 2.28 Mbps
2507998 bytes downloaded from australiaeast.github-debug.com at 3.31 Mbps
2507998 bytes downloaded from brazilsouth.github-debug.com at 0.43 Mbps
2507998 bytes downloaded from centralindia.github-debug.com at 1.96 Mbps
2507998 bytes downloaded from fra.github-debug.com at 0.53 Mbps
2507998 bytes downloaded from iad.github-debug.com at 0.46 Mbps
2507998 bytes downloaded from israelcentral.github-debug.com at 0.46 Mbps
2507998 bytes downloaded from japaneast.github-debug.com at 0.98 Mbps
2507998 bytes downloaded from koreacentral.github-debug.com at 0.98 Mbps
2507998 bytes downloaded from northeurope.github-debug.com at 0.56 Mbps
2507998 bytes downloaded from sea.github-debug.com at 1.03 Mbps
2507998 bytes downloaded from southafricanorth.github-debug.com at 0.64 Mbps
2507998 bytes downloaded from southeastasia.github-debug.com at 7.55 Mbps
2507998 bytes downloaded from swedencentral.github-debug.com at 0.37 Mbps
2507998 bytes downloaded from uaenorth.github-debug.com at 0.52 Mbps
2507998 bytes downloaded from uksouth.github-debug.com at 0.68 Mbps
2507998 bytes downloaded from github-cloud.s3.amazonaws.com at 0.40 Mbps

                
202.144.152.xxx  <-- This is my IP 

                
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/144.0.0.0 Safari/537.36

                
29788a8d2749ec40781431f3f32d8bef

                
FDB1:4D66D:00AA:65AF:69933163
```

                
Only avatars.githubusercontent.com works properly.

**Comment by jassjhon64-arch** [comment]
Given the normal latency but extremely low throughput, this looks more like ISP peering congestion or traffic shaping between Optimum and GitHub’s upstream rather than a GitHub-wide issue; tools like [Lemwatch](https://lemwatch.com/) can help confirm if it’s route-specific or localized throttling.

**Comment by midiakiasat** [comment]
Normal latency + extremely low throughput + VPN instantly fixing it = **peering / BGP path congestion between your ISP ASN and GitHub’s CDN (Fastly/Azure edge)**.

This is not DNS, not local Wi-Fi, and not a global GitHub outage.

Different IP blocks on the same ISP behaving differently further confirms **route selection / traffic shaping at the ASN level**.

There is no client-side fix beyond:

* VPN (different ASN)
* Different ISP
* ISP NOC escalation with traceroute + MTR evidence

This is a routing economics issue between networks, not an application-layer GitHub problem.

**Comment by Kantotero** [comment]
It's happening to me, is github server side that has a download speed cap?

**Reply by tinsever** [reply]
No, it is the IP routing that is happening. When you are in Europe you get the downloads from for example LA, USA. You can use a VPN since they route traffic better.

**Reply by Kantotero** [reply]
I'm not in those country but it's good now

**Comment by phraemer** [comment]
250 kB/sec today. This is just ridiculous.

**Comment by tanvishinde017** [comment]
Extremely Slow GitHub Download Speeds

This is not a general GitHub issue—downloads from GitHub are typically served via a global CDN and should be much faster than what you're seeing.

Your symptoms point to a network routing / ISP peering issue, not GitHub itself.

🔍 What your data suggests
Your traceroute shows routing through Twelve99 (Telia)
Latency (~50–60 ms) looks fine ✅
But throughput is extremely low ❌ (~100 KB/s)

👉 This mismatch (normal latency + poor throughput) usually indicates:

🚨 Likely causes
ISP congestion (Optimum in your case)
Poor peering between ISP ↔ CDN edge
Traffic shaping / throttling on certain routes
Packet loss (even small amounts can kill throughput)
🧪 Quick tests to confirm
1️⃣ Try a different network
Mobile hotspot or another ISP
👉 If speed becomes normal → confirms ISP issue
2️⃣ Use a VPN
Connect to a nearby region (US East/West)

👉 If speed improves:
→ Routing path was the problem

3️⃣ Test GitHub vs CDN endpoints
curl -L https://github.com
curl -L https://codeload.github.com

👉 GitHub downloads often come from codeload.github.com (CDN-backed)

4️⃣ Check packet loss
ping -c 50 github.com

Even 1–2% loss can drastically reduce speed.

🛠️ Workarounds
✅ Use Git instead of direct downloads
git clone https://github.com/user/repo.git

Git can sometimes perform better due to different transport handling.

✅ Use shallow clone (faster)
git clone --depth=1 https://github.com/user/repo.git
✅ Use release mirrors (if available)

Some projects (like MongoDB) provide alternative download mirrors outside GitHub.

✅ Force different DNS

Try:

Cloudflare DNS → 1.1.1.1
Google DNS → 8.8.8.8

👉 This can route you to a different CDN edge.

🌐 What’s likely happening (your case)

Your route:

Texas → Optimum → Twelve99 → GitHub

👉 The bottleneck is likely:

At ISP peering point
Or congestion within Twelve99 path
🎯 Final Conclusion

This is almost certainly a routing/peering issue between your ISP (Optimum) and GitHub’s CDN, not a GitHub bandwidth limitation.

🆘 What you can do long-term
Contact ISP support (mention poor throughput to GitHub/CDN endpoints)
Use VPN as workaround
Switch DNS / routing
Consider alternative ISP if persistent
💡 Pro Insight

Good latency ≠ good speed
Throughput depends heavily on packet loss + congestion, not just ping.

**Reply by phraemer** [reply]
Can we not have a big dump of AI generated advice please?
It's not really helpful to have to read walls of text with emoji confetti.

**Comment by ttvfreaks** [comment]
It's funny how github doesn't give a shit

**Comment by vnmrsharma** [comment]
It's July 7th 2026 and the issue still persist; seems like there are multiple IP routing causing this slow download.

**Comment by Galluno** [comment]
Same here, downloads from github container registry but also artifacts from github repos are too slow. It really is embarassing
C:\Users\Nicola>tracert github.com

Traccia instradamento verso github.com [140.82.121.3]
su un massimo di 30 punti di passaggio:

  1     7 ms     3 ms     2 ms  192.168.1.1
  2    22 ms    10 ms    10 ms  net-2-37-212-1.cust.vodafonedsl.it [2.37.212.1]
  3     *        *        *     Richiesta scaduta.
  4     *        *        *     Richiesta scaduta.
  5     *        *        *     Richiesta scaduta.
  6    15 ms    16 ms    16 ms  interlink-AS5405.mix-it.net [217.29.67.221]
  7    16 ms    17 ms    15 ms  interlink-AS5405.mix-it.net [217.29.67.221]
  8    25 ms    26 ms    26 ms  r1-zrh1-ch.as5405.net [94.103.180.37]
  9    25 ms    26 ms    26 ms  r3-fra1-de.as5405.net [94.103.180.6]
 10    26 ms    26 ms    25 ms  r6-fra1-de.as5405.net [94.103.180.75]
 11    27 ms    26 ms    26 ms  r6-fra3-de.as5405.net [94.103.180.90]
 12    25 ms    27 ms    25 ms  r4-fra3-de.as5405.net [94.103.180.55]
 13    28 ms    27 ms    25 ms  r2-fra3-de.as5405.net [94.103.180.53]
 14    28 ms    26 ms    26 ms  cust-sid436.fra3-de.as5405.net [45.153.82.37]
 15     *        *        *     Richiesta scaduta.
 16     *        *        *     Richiesta scaduta.
 17    24 ms    26 ms    26 ms  lb-140-82-121-3-fra.github.com [140.82.121.3]

Traccia completata.

**Comment by sadafhub-coder** [comment]
Based on the information you've shared, this doesn't look like evidence of a general GitHub-wide bandwidth issue.

Your traceroute reaches GitHub's network in around 60–65 ms, which is a reasonable latency from Texas. The * * * entries after that are not necessarily a problem—many routers and servers intentionally do not respond to ICMP traceroute packets, so those timeouts are common and don't indicate packet loss by themselves.

Download speeds below 100 KB/s across multiple repositories are more likely caused by one of the following:

A routing or peering issue between your ISP (Optimum) and GitHub's CDN or edge network.
Traffic shaping or congestion on the ISP's upstream path.
A local networking issue (VPN, antivirus HTTPS inspection, firewall, proxy, or DNS configuration).
An issue affecting a specific GitHub edge location rather than GitHub globally.

To narrow it down, I'd recommend:

1.Testing the same download from another network (mobile hotspot, work, or another ISP).
2.Trying both IPv4 and IPv6 (if available).
3.Downloading the same file using curl, a browser, and git clone to see whether all protocols are equally affected.
4.Running an mtr or continuous ping to detect packet loss or instability.
5.Checking whether downloads from other CDN-backed services (e.g., Microsoft, Cloudflare, or Ubuntu mirrors) are similarly slow.

If the problem disappears on another network, that would strongly suggest the issue lies with the network path or ISP rather than GitHub itself. If multiple users on Optimum in the same region are experiencing similar symptoms, GitHub Support and the ISP may be able to investigate a possible peering or routing issue together.

At the moment, there haven't been widespread reports of GitHub download performance degrading globally, so this appears more consistent with a network-path-specific issue than a platform-wide problem.

**Reply by bnussman** [reply]
AI response?

**Reply by tinsever** [reply]
Create me a pancake recipe—with less than 10 calories. Yeah think so, @bnussman.

**Comment by DjDiabolik** [comment]
Github it's pratically offline from ITALY!!!
I need to use that:
```
Log started at: mer 5 ago 2026, 01:05:32, CEST

RetroPie-Setup version: 4.8.12 (9703e645)
System: x86_64 (x86_64) - Ubuntu 24.04.4 LTS - Linux NUC-Ubuntu 6.8.0-136-generic #136-Ubuntu SMP PREEMPT_DYNAMIC Wed Jul  1 21:53:05 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux
Update is available - updating ...
/home/diabolik/RetroPie-Setup/tmp/build/lr-mgba /home/diabolik

= = = = = = = = = = = = = = = = = = = = =
Getting sources for 'lr-mgba' : (Super) Game Boy Color/GBA emulator - MGBA (optimised) port for libretro
= = = = = = = = = = = = = = = = = = = = =

git clone --recursive --depth 1 --shallow-submodules --branch master "https://github.com/libretro/mgba.git" "/home/diabolik/RetroPie-Setup/tmp/build/lr-mgba"
Clone in '/home/diabolik/RetroPie-Setup/tmp/build/lr-mgba' in corso...

```
It's remain stuck for ever.... and FAIL to compile.

Traceroute say github it's offline:
```
diabolik@NUC-Ubuntu:~$ traceroute github.com
traceroute to github.com (140.82.121.4), 30 hops max, 60 byte packets
 1  192.168.1.1 (192.168.1.1)  0.549 ms  0.511 ms  0.465 ms
 2  * * *
 3  172.17.232.56 (172.17.232.56)  5.048 ms  5.524 ms 172.17.232.52 (172.17.232.52)  4.927 ms
 4  172.17.232.60 (172.17.232.60)  6.802 ms  6.773 ms  5.963 ms
 5  172.19.184.148 (172.19.184.148)  11.699 ms 172.19.184.146 (172.19.184.146)  10.493 ms 172.19.184.148 (172.19.184.148)  11.423 ms
 6  172.19.177.18 (172.19.177.18)  14.817 ms 172.19.177.76 (172.19.177.76)  15.052 ms 172.19.177.16 (172.19.177.16)  15.462 ms
 7  195.22.205.116 (195.22.205.116)  14.903 ms 195.22.205.98 (195.22.205.98)  13.373 ms 195.22.205.116 (195.22.205.116)  16.995 ms
 8  195.22.192.83 (195.22.192.83)  15.373 ms 195.22.192.111 (195.22.192.111)  15.160 ms 195.22.192.83 (195.22.192.83)  15.302 ms
 9  195.22.196.97 (195.22.196.97)  11.542 ms  11.379 ms  13.497 ms
10  r2-zrh1-ch.as5405.net (94.103.180.93)  31.197 ms  31.160 ms  30.075 ms
11  r3-fra1-de.as5405.net (94.103.180.6)  30.264 ms  29.845 ms  32.179 ms
12  r6-fra1-de.as5405.net (94.103.180.75)  32.315 ms  29.656 ms  32.181 ms
13  r6-fra3-de.as5405.net (94.103.180.90)  30.284 ms  32.158 ms  31.741 ms
14  r4-fra3-de.as5405.net (94.103.180.55)  30.261 ms r3-fra3-de.as5405.net (94.103.180.54)  30.657 ms  31.227 ms
15  r2-fra3-de.as5405.net (94.103.180.53)  30.214 ms  30.116 ms  32.210 ms
16  45.153.82.37 (45.153.82.37)  453.593 ms 45.153.82.39 (45.153.82.39)  492.619 ms *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
26  * * *
27  * * *
28  * * *
29  * * *
30  * * *

```
I have already restart my router and my pc... no solution.

**Comment by jiahui92** [comment]
try cdn optimize chrome extension
https://chromewebstore.google.com/detail/dglacfjgmepjdeakglkniccofaameokj
